clar2wasm/src/tools.rs

```diff
@@ -430,18 +421,31 @@ pub fn crosscheck_validate<V: Fn(Value)>(snippet: &str, validator: V) {
     let interpreted = interpret(snippet);
 
     assert_eq!(
-        compiled.as_ref().map_err(|_| &()),
-        interpreted.as_ref().map_err(|_| &()),
+        compiled, interpreted,
         "Compiled and interpreted results diverge! {}\ncompiled: {:?}\ninterpreted: {:?}",
-        snippet,
-        &compiled,
-        &interpreted
+        snippet, &compiled, &interpreted
     );
 
     let value = compiled.unwrap().unwrap();
     validator(value)
 }
 
+// TODO: After issue #421 is complete,
+// several tests that call this function will need to be adjusted.
+pub fn crosscheck_expect_failure(snippet: &str) {
+    let compiled = evaluate(snippet);
+    let interpreted = interpret(snippet);
+
+    assert_eq!(
+        compiled.is_err(),
+        interpreted.is_err(),
+        "Compiled and interpreted results diverge! {}\ncompiled: {:?}\ninterpreted: {:?}",
+        snippet,
+        &compiled,
+        &interpreted
+    );
+}
```

Comment:
This function is not a good idea for the same reason as `crosscheck_compare_only` should never be used.

Let's imagine that I want to check for the underflow of uints substraction and write this:
```rust
#[test]
fn uint_underflow() {
    crosscheck_expect_failure("(- u0 u1");
}
```
Here the test will pass but for the wrong reason: it's a syntax error.

I think it's extremely important to make sure we have the expected error.

Reply:
That function's purpose is to be used when compilation errors are thrown, specifically to avoid marking tests with IGNORE and TODO notes. It should be removed once https://github.com/stacks-network/clarity-wasm/issues/421 is implemented.